core/src/main/java/com/linecorp/armeria/common/HttpRequest.java

```diff
@@ -246,8 +242,8 @@ static HttpRequest of(RequestHeaders headers, HttpData content, HttpHeaders trai
      * {@code objs} must not contain {@link RequestHeaders}.
      */
     static HttpRequest of(RequestHeaders headers, HttpObject... objs) {
-        if (Arrays.stream(objs).anyMatch(obj -> obj instanceof HttpHeaders)) {
-            throw new IllegalArgumentException("objs contains Headers, which is not allowed.");
+        if (Arrays.stream(objs).anyMatch(obj -> obj instanceof RequestHeaders)) {
```

Comment:
`!(obj instanceof HttpData)`?

Reply:
The last element of `objs` could be `HttpHeaders` which is an HTTP trailers.
Do we need to change the API to prohibit it?

Reply:
Or check the last element separately?

Reply:
I think the check will be needed for response, but for request we can prohibit it using @trustin's check since they're not allowed as mentioned above.

Reply:
Ah right, I was totally misunderstanding. :-)

Reply:
Hmm... Why don't we just accept `HttpData...` at the first place? (and update Javadoc)

Reply:
Yeah I was making the API deprecated and adding a new one. :-)

Reply:
Realize that does not have to deprecate the old one and add new one becuase users would not notice the difference.